### PR TITLE
JWT auth

### PR DESCRIPTION
**Pull request by dholms**
Added simple JWT auth to the server. 

- Encodes the user's `did` as the `sub` (subject) & signs w `HS256`
- Added passwords to users table in DB, hashed w bcrypt
- stuck auth on all of the current authed routes
- fleshed out `createSession` for logging in & receiving a jwt & `getSession` for getting current user info with an authed request
- no expiration date on JWTs yet. not sure what a reasonable time frame would be 🤔 a week? a month?

### CODE REVIEW

File: packages/lex-cli/src/codegen/client.ts [hunk 1]
```diff
@@ -146,6 +146,23 @@ const indexTs = (project: Project, schemas: Schema[], nsidTree: NsidNS[]) =>
         ].join('\n'),
       )
 
+    //= setHeader(key: string, value: string): void {
+    //=   this.xrpc.setHeader(key, value)
+    //= }
+    const setHeaderMethod = serviceClientCls.addMethod({
```
Comment (dholms) [inline packages/lex-cli/src/codegen/client.ts:152]: new method. make sure i didn't do anything silly 😛

File: packages/lex-cli/src/codegen/client.ts [hunk 2]
```diff
@@ -327,10 +344,14 @@ function genRecordCls(file: SourceFile, schema: RecordSchema) {
       name: 'record',
       type: `${typeModule}.Record`,
     })
+    method.addParameter({
+      name: 'headers?',
```
Comment (dholms) [inline packages/lex-cli/src/codegen/client.ts:348]: added optional `headers` param on record methods

File: packages/lex-cli/src/codegen/client.ts (hunk 1 above)
Comment (pfrazee) [reply packages/lex-cli/src/codegen/client.ts:152]: Nailed it

File: packages/lex-cli/src/codegen/client.ts (hunk 2 above)
Comment (pfrazee) [reply packages/lex-cli/src/codegen/client.ts:348]: 👍

File: packages/server/src/db/index.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
+import { DataSource } from 'typeorm'
+import * as bcrypt from 'bcryptjs'
```
Comment (pfrazee) [inline packages/server/src/db/index.ts:2]: Why use a JS implementation when node has a native one?

**Review by pfrazee — APPROVED** [review]
Looks good, just one question about the bcrypt library

File: packages/server/src/db/index.ts (hunk 3 above)
Comment (dholms) [reply packages/server/src/db/index.ts:2]: oh node has native scrypt 🤔 
https://nodejs.org/api/crypto.html#crypto_crypto_scrypt_password_salt_keylen_options_callback

Comment (pfrazee) [reply packages/server/src/db/index.ts:2]: oh my b. Is bcrypt that much better? Just gonna suck resources to do it in js

Comment (dholms) [reply packages/server/src/db/index.ts:2]: no looks like scrypt is actually better. ill switch it over 👌

**Pull request merged**